src/telemetry.ts

```diff
@@ -1,137 +1,200 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
 import * as vscode from 'vscode';
 import TelemetryReporter from 'vscode-extension-telemetry';
 
+import {CancelOperationError} from './CancelOperationError';
 import {DevelopEnvironment} from './constants';
 import {ExceptionHelper} from './exceptionHelper';
 import {RemoteExtension} from './Models/RemoteExtension';
 import {NSAT} from './nsat';
-import {InternalConfig} from './utils';
+import {WorkbenchExtension} from './WorkbenchExtension';
+
 
-interface PackageInfo {
+interface ExtensionInfo {
   name: string;
   version: string;
   aiKey: string;
 }
 
-
-export interface TelemetryContext {
-  properties: TelemetryProperties;
-  measurements: TelemetryMeasurements;
+/**
+ * Operation result of telemetry
+ */
+export enum TelemetryResult {
+  Succeeded = 'Succeeded',
+  Failed = 'Failed',
+  Cancelled = 'Cancelled',
 }
 
-export interface TelemetryProperties {
-  result: 'Succeeded'|'Failed'|'Cancelled';
-  error: string;
-  errorMessage: string;
-  [key: string]: string;
+/**
+ * Context of telemetry
+ */
+export interface TelemetryContext {
+  properties: {[key: string]: string};
+  measurements: {[key: string]: number};
 }
 
-export interface TelemetryMeasurements {
-  duration: number;
-  [key: string]: number;
-}
+export class TelemetryWorker {
+  private _reporter: TelemetryReporter|undefined;
+  private extensionContext: vscode.ExtensionContext|undefined;

```

Comment:
Please use consistent naming convention for the private field.

Reply:
Fixed.